dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/CompanyUploader.kt

```diff
@@ -15,34 +18,53 @@ import java.net.SocketTimeoutException
 @Service
 class CompanyUploader(
     @Autowired private val companyDataControllerApi: CompanyDataControllerApi,
+    @Autowired private val objectMapper: ObjectMapper,
 ) {
     companion object {
         const val MAX_RETRIES = 3
         const val UNAUTHORIZED_CODE = 401
-        const val FORBIDDEN_CODE = 403
     }
 
     private val logger = LoggerFactory.getLogger(javaClass)
 
-    /**
-     * Uploads a single Company to the dataland backend performing at most MAX_RETRIES retries.
-     * This function absorbs errors and logs them.
-     */
-    fun uploadSingleCompany(
-        companyInformation: CompanyInformation,
-    ) {
+    @Suppress("ReturnCount")
+    private fun checkForDuplicateIdentifierAndGetConflictingCompanyId(exception: ClientException): String? {
+        if (exception.statusCode != HttpStatus.BAD_REQUEST.value()) {
+            return null
+        }
+
+        val exceptionResponse = exception.response!! as ClientError<*>
+        val exceptionBodyString = exceptionResponse.body.toString()
+
+        val errorResponseBody = objectMapper.readTree(exceptionBodyString)
+        val firstError = errorResponseBody.get("errors").get(0)
+        if (firstError.get("errorType").textValue() != "duplicate-company-identifier") {
+            return null
+        }
+
+        val conflictingIdentifiers = firstError.get("metaInformation")
+        if (!conflictingIdentifiers.isArray || conflictingIdentifiers.size() != 1) {
+            return null
+        }
+
+        val conflictingIdentifier = conflictingIdentifiers.get(0)
+        val conflictingIdentifierType = conflictingIdentifier.get("identifierType").textValue()
+        if (conflictingIdentifierType != "Lei") {
+            return null
+        }
+
+        return conflictingIdentifier.get("companyId").textValue()
+    }
+
+    private fun retryCatchingErrors(function: () -> Unit) {
         var counter = 0
         while (counter < MAX_RETRIES) {
             try {
-                logger.info(
-                    "Uploading company data for ${companyInformation.companyName} " +
-                        "(LEI: ${companyInformation.identifiers["Lei"]!!.first()})",
-                )
-                companyDataControllerApi.postCompany(companyInformation)
+                function()
                 break
             } catch (exception: ClientException) {
                 logger.error("Unable to upload company data. Response was: ${exception.message}.")
```

Comment:
Should we change this log message? Since the method gets an unknown function, we do not know here if the upload of company data failed.

Reply:
Changed to match the log message used in other cases.